### PR TITLE
DarkTheme Schedules/Payees

### PR DESCRIPTION
**Pull request by carkom**
Pulling in Schedules/Payees as the next phase of the darkTheme integration.

### CODE REVIEW

**Comment by carkom** [comment]
@MatissJanis this one is ready for review.

File: packages/desktop-client/src/components/SidebarWithData.js [hunk 1]
```diff
@@ -79,8 +79,8 @@ function EditableBudgetName({ prefs, savePrefs }) {
     return (
       <Button
         type="bare"
-        color={colors.n9}
         style={{
+          color: theme.sidebarItemText,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/SidebarWithData.js:83]: **🔨 warning:** ‏the color is wrong on hover

Screenshot 2023-08-12 at 09 19 46

File: packages/desktop-client/src/components/manager/subscribe/ChangePassword.tsx [hunk 2]
```diff
@@ -60,7 +60,8 @@ export default function ChangePassword() {
         <Text
           style={{
             marginTop: 20,
-            color: colors.r4,
+            color: theme.errorText,
+            backgroundColor: theme.errorBackground,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/manager/subscribe/ChangePassword.tsx:64]: **🔨 warning:** ‏I don't think we need this background color.

Screenshot 2023-08-12 at 09 22 52

File: packages/desktop-client/src/components/manager/subscribe/ChangePassword.tsx [hunk 3]
```diff
@@ -73,7 +74,8 @@ export default function ChangePassword() {
         <Text
           style={{
             marginTop: 20,
-            color: colors.g4,
+            color: theme.noticeText,
+            backgroundColor: theme.noticeBackground,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/manager/subscribe/ChangePassword.tsx:78]: Same comment here

File: packages/desktop-client/src/components/manager/subscribe/Bootstrap.tsx [hunk 4]
```diff
@@ -74,7 +74,8 @@ export default function Bootstrap() {
         <Text
           style={{
             marginTop: 20,
-            color: colors.r4,
+            color: theme.errorText,
+            backgroundColor: theme.errorBackground,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/manager/subscribe/Bootstrap.tsx:78]: Same comment here

File: packages/desktop-client/src/components/schedules/EditSchedule.js [hunk 5]
```diff
@@ -487,7 +487,7 @@ export default function ScheduleDetails({ modalProps, actions, id }) {
               }}
               style={{
                 padding: '0 10px',
-                color: colors.n5,
+                color: theme.pageText,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/EditSchedule.js:490]: **🔨 warning:** ‏wrong color

Before:
Screenshot 2023-08-12 at 09 28 07


After:
Screenshot 2023-08-12 at 09 28 12

File: packages/desktop-client/src/components/schedules/EditSchedule.js [hunk 6]
```diff
@@ -624,7 +618,6 @@ export default function ScheduleDetails({ modalProps, actions, id }) {
             style={{
               width: 350,
               textAlign: 'right',
-              color: colors.n4,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/EditSchedule.js:627]: **🔨 warning:** ‏wrong color; also the text gets broken up incorrectly (when comparing against v23.8.1)

Before:
Screenshot 2023-08-12 at 09 29 10


After:
Screenshot 2023-08-12 at 09 29 14

File: packages/desktop-client/src/components/schedules/EditSchedule.js [hunk 7]
```diff
@@ -547,14 +547,8 @@ export default function ScheduleDetails({ modalProps, actions, id }) {
...
-              <Text style={{ color: colors.n4, fontWeight: 600 }}>
-                Upcoming dates
-              </Text>
-              <Stack
-                direction="column"
-                spacing={1}
-                style={{ marginTop: 10, color: colors.n4 }}
-              >
+              <Text style={{ fontWeight: 600 }}>Upcoming dates</Text>
+              <Stack direction="column" spacing={1} style={{ marginTop: 10 }}>
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/EditSchedule.js:551]: **🔨 warning:** ‏wrong colors

before:
Screenshot 2023-08-12 at 09 30 21


after:
Screenshot 2023-08-12 at 09 30 19

File: packages/desktop-client/src/components/schedules/EditSchedule.js [hunk 8]
```diff
@@ -674,7 +667,9 @@ export default function ScheduleDetails({ modalProps, actions, id }) {
                 type="bare"
                 style={{
                   color:
-                    state.transactionsMode === 'linked' ? colors.b4 : colors.n7,
+                    state.transactionsMode === 'linked'
+                      ? theme.tableTextEditingBackground
+                      : 'inherit',
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/EditSchedule.js:672]: **🔨 warning:** ‏wrong color

before
Screenshot 2023-08-12 at 09 31 38


after
Screenshot 2023-08-12 at 09 31 35

File: packages/desktop-client/src/components/schedules/SchedulesTable.js [hunk 9]
```diff
@@ -93,7 +93,7 @@ export function ScheduleAmountCell({ amount, op }) {
         <View
           style={{
             textAlign: 'left',
-            color: colors.n7,
+            color: theme.tableText,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/SchedulesTable.js:96]: **🔨 warning:** ‏wrong color

before
Screenshot 2023-08-12 at 09 33 03


after
Screenshot 2023-08-12 at 09 33 01

File: packages/desktop-client/src/components/schedules/SchedulesTable.js [hunk 10]
```diff
@@ -192,17 +192,21 @@ export function SchedulesTable({
...
         inset={15}
-        backgroundColor="transparent"
         onClick={() => onSelect(item.id)}
         style={{
           cursor: 'pointer',
-          backgroundColor: 'white',
-          ':hover': { backgroundColor: colors.hover },
+          backgroundColor: theme.tableBackground,
+          color: theme.tableText,
+          ':hover': { backgroundColor: theme.tableRowBackgroundHover },
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/SchedulesTable.js:200]: **🔨 warning:** 
- [ ] the table is missing `:hover` effects
- [ ] the content text color is different
- [ ] "show completed schedules" color is different

File: packages/desktop-client/src/components/schedules/index.js [hunk 11]
```diff
@@ -83,7 +83,6 @@ export default function Schedules() {
           allowCompleted={true}
           onSelect={onEdit}
           onAction={onAction}
-          style={{ backgroundColor: 'white' }}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/index.js:86]: **🔨 warning:** ‏oops?

Before: empty schedule table has white background

After: empty schedule table has gray-ish background

File: packages/desktop-client/src/components/select/DateSelect.js [hunk 12]
```diff
@@ -32,8 +32,8 @@ import DateSelectRight from './DateSelect.right.png';
 
 let pickerStyles = {
   '& .pika-single.actual-date-picker': {
-    color: colors.n11,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/select/DateSelect.js:35]: **🔨 warning:** ‏date selector is **significantly** different.

File: packages/desktop-client/src/components/sidebar.js [hunk 13]
```diff
@@ -446,13 +447,14 @@ function Accounts({
...
     <View className="float" style={[style, { flexShrink: 0 }]}>
-      <Button type="bare" onClick={onFloat} color={colors.n5}>
+      <Button type="bare" onClick={onFloat}>
         {isFloating ? (
           <Pin
             style={{
               margin: -2,
               width: 15,
               height: 15,
+              color: theme.altpageTextSubdued,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar.js:457]: **🔨 warning:** ‏color of the collapse/expand button is wrong

Screenshot 2023-08-12 at 09 44 28

File: packages/desktop-client/src/components/schedules/DiscoverSchedules.js [hunk 14]
```diff
@@ -91,8 +98,6 @@ function DiscoverSchedulesTable({ schedules, loading }) {
       </TableHeader>
       <Table
         rowHeight={ROW_HEIGHT}
-        version="v2"
-        backgroundColor="transparent"
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/DiscoverSchedules.js:95]: **🔨 warning:** ‏this table is missing a `:hover` row highlight.

File: packages/desktop-client/src/components/payees/index.js [hunk 15]
```diff
@@ -544,7 +548,7 @@ export const ManagePayees = forwardRef(
           <View
             style={{
               flex: 1,
-              border: '1px solid ' + colors.border,
+              border: '1px solid ' + theme.tableBorder,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/payees/index.js:551]: **🔨 warning:** ‏the payee table has quite a few changes too

- [ ] "create rule" button background color is wrong
- [ ] table is missing row `:hover` effect
- [ ] table content color is different
- [ ] status texts have different color (i.e. search for some payee that does not exist to see "no payee" text)
- [ ] selected payee text color is wrong (screenshot below)


before:
Screenshot 2023-08-12 at 09 48 51


after:
Screenshot 2023-08-12 at 09 48 22

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Some other changes I'm noticing:

 - [ ] the filter operator pills now have different background (gray vs blue)
 - [ ] rules: the "pre"/"default"/"post" pills have wrong background color

**Comment by MatissJanis** [comment]
Heads up: I've fixed some of the table `:hover` effects. You might just need to pull in the latest changes from master to see them fixed here too.

File: packages/desktop-client/src/components/SidebarWithData.js (hunk 1 above)
Comment (carkom) [reply packages/desktop-client/src/components/SidebarWithData.js]: Fixed

File: packages/desktop-client/src/components/schedules/EditSchedule.js (hunk 6 above)
Comment (carkom) [reply packages/desktop-client/src/components/schedules/EditSchedule.js]: This formatting is an issue with the current master and is not a result of anything within this PR.

I have fixed the colors.

File: packages/desktop-client/src/components/payees/index.js (hunk 15 above)
Comment (carkom) [reply packages/desktop-client/src/components/payees/index.js:546]: All fixed.

**Comment by MatissJanis** [comment]
FYI: I just added some more VRT tests. When you pull in the latest changes from `master` into this branch - the VRT ci job should fail now.

**Comment by carkom** [comment]
@MatissJanis , can we get this one reviewed and merged? I'm spending a lot of time fixing and refixing new merges that are not following theme standards, eg. using colors instead of theme. The longer this takes to merge the more work it becomes.

**Comment by MatissJanis** [comment]
Visual regression tests are failing. Please look at fixing the issues these tests caught. After all CI jobs pass I'll do another round of manual reviews and hopefully won't spot any new issues and then we'll be able to merge.

**Comment by carkom** [comment]
> Visual regression tests are failing. Please look at fixing the issues these tests caught. After all CI jobs pass I'll do another round of manual reviews and hopefully won't spot any new issues and then we'll be able to merge.

I like that VRT test! Makes this work much easier. All set, thanks!

File: packages/desktop-client/src/components/schedules/EditSchedule.js [hunk 16]
```diff
@@ -419,7 +419,7 @@ export default function ScheduleDetails({ modalProps, actions, id }) {
       size="medium"
       {...modalProps}
     >
-      <Stack direction="row" style={{ marginTop: 10 }}>
+      <Stack direction="row" style={{ marginTop: 10, color: theme.pageText }}>
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/EditSchedule.js:422]: Same question

File: packages/desktop-client/src/components/schedules/StatusBadge.tsx [hunk 17]
```diff
@@ -9,62 +9,71 @@ import CheckCircleHollow from '../../icons/v2/CheckCircleHollow';
...
 
 export function getStatusProps(status: Status) {
-  let color, backgroundColor, Icon;
+  let color, backgroundColor, borderColor, Icon;
 
   switch (status) {
     case 'missed':
-      color = colors.r1;
-      backgroundColor = colors.r10;
+      color = theme.errorText;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/StatusBadge.tsx:21]: **🔨 warning:** ‏some of these colors are quite different. I'm expanding the VRT tests to catch this in the future: https://github.com/actualbudget/actual/pull/1672

(yes, I know they are inconsistent with the list-view; I think the list view color change is a recent regression.. either way - we can address this independently - lets try and keep the existing lightmode with no changes unless absolutelly necessary)

File: packages/desktop-client/src/components/payees/index.js [hunk 18]
```diff
@@ -220,22 +222,20 @@ const PayeeTable = forwardRef(
...
 
 function PayeeTableHeader() {
-  let borderColor = colors.border;
   let dispatchSelected = useSelectedDispatch();
   let selectedItems = useSelectedItems();
 
   return (
     <View>
       <TableHeader
+        borderColor={theme.tableborder}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/payees/index.js:231]: **🔨 warning:** ‏`TableHeader` does not have a `borderColor` prop

File: packages/desktop-client/src/components/sidebar/SidebarWithData.tsx [hunk 19]
```diff
@@ -88,6 +88,7 @@ function EditableBudgetName({ prefs, savePrefs }: EditableBudgetNameProps) {
         type="bare"
         color={theme.buttonNormalBorder}
         style={{
+          color: theme.sidebarItemText,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/SidebarWithData.tsx:91]: **❓ question:** ‏why is this necessary?

File: packages/desktop-client/src/components/schedules/SchedulesTable.tsx [hunk 20]
```diff
@@ -226,7 +231,7 @@ export function SchedulesTable({
         {!minimal && (
           <Field width={80} style={{ textAlign: 'center' }}>
             {item._date && item._date.frequency && (
-              <Check style={{ width: 13, height: 13 }} />
+              <Check style={{ width: 13, height: 13, color: 'inherit' }} />
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/SchedulesTable.tsx:234]: **❓ question:** ‏is this necessary?

File: packages/desktop-client/src/components/select/DateSelect.js [hunk 21]
```diff
@@ -45,28 +45,33 @@ let pickerStyles = {
...
+      backgroundColor: theme.sidebarItemBackgroundHover,
+      color: theme.sidebarItemText,
     },
     '& .is-today .pika-button': {
       textDecoration: 'underline',
     },
     '& .is-selected .pika-button': {
-      backgroundColor: colors.n5,
+      backgroundColor: theme.buttonNormalSelectedBackground,
+      color: theme.buttonNormalSelectedText,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/select/DateSelect.js:74]: **🔨 warning:** ‏wrong color on lightmode

File: packages/desktop-client/src/components/select/RecurringSchedulePicker.js [hunk 22]
```diff
@@ -172,7 +172,9 @@ function SchedulePreview({ previewDates }) {
   } else {
     content = (
       <View>
-        <Text style={{ fontWeight: 600 }}>Upcoming dates</Text>
+        <Text style={{ color: theme.tableText, fontWeight: 600 }}>
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/select/RecurringSchedulePicker.js:175]: **❓ question:** ‏why do we need to set the color here?

File: packages/desktop-client/src/components/sidebar/Account.tsx [hunk 23]
```diff
@@ -95,16 +94,16 @@ function Account({
               ...(updated && { fontWeight: 700 }),
             }}
             activeStyle={{
-              borderColor: colors.p8,
-              color: colors.p8,
+              borderLeft: '4px solid ' + theme.sidebarItemAccentSelected,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/Account.tsx:97]:
```suggestion
              borderColor: theme.sidebarItemAccentSelected,
```

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Sorry for being so nitpicky, but last time we merged a darkmode PR there were a lot of regressions that I had to fix. I don't want to repeat the same mistake so taking extra care here + improving our VRT coverage for issues I spot.

File: packages/desktop-client/src/components/manager/subscribe/Bootstrap.tsx [hunk 24]
```diff
@@ -54,14 +54,14 @@ export default function Bootstrap() {
   }
 
   return (
-    <View style={{ maxWidth: 450, marginTop: -30 }}>
+    <View style={{ maxWidth: 450, marginTop: -30, color: theme.pageText }}>
```
[the remark this answers is not in the document]
Comment (carkom) [reply packages/desktop-client/src/components/manager/subscribe/Bootstrap.tsx:57]: Removed

**Comment by carkom** [comment]
> Sorry for being so nitpicky, but last time we merged a darkmode PR there were a lot of regressions that I had to fix. I don't want to repeat the same mistake so taking extra care here + improving our VRT coverage for issues I spot.

Yep, totally get it. I agree with your sentiment. Sorry for all the unneccary changes in the code. I'll scrub the next ones before putting forward for review.

File: packages/desktop-client/src/components/manager/subscribe/ChangePassword.tsx [hunk 25]
```diff
@@ -43,12 +43,12 @@ export default function ChangePassword() {
   }
 
   return (
-    <View style={{ maxWidth: 500, marginTop: -30 }}>
+    <View style={{ maxWidth: 500, marginTop: -30, color: theme.pageText }}>
```
[the remark this answers is not in the document]
Comment (carkom) [reply packages/desktop-client/src/components/manager/subscribe/ChangePassword.tsx:46]: Removed

File: packages/desktop-client/src/components/schedules/EditSchedule.js (hunk 16 above)
Comment (carkom) [reply packages/desktop-client/src/components/schedules/EditSchedule.js:422]: Removed

File: packages/desktop-client/src/components/sidebar/SidebarWithData.tsx (hunk 19 above)
Comment (carkom) [reply packages/desktop-client/src/components/sidebar/SidebarWithData.tsx:91]: Removed

File: packages/desktop-client/src/components/schedules/SchedulesTable.tsx (hunk 20 above)
Comment (carkom) [reply packages/desktop-client/src/components/schedules/SchedulesTable.tsx:234]: Removed

File: packages/desktop-client/src/components/schedules/StatusBadge.tsx (hunk 17 above)
Comment (carkom) [reply packages/desktop-client/src/components/schedules/StatusBadge.tsx:21]: Okay, I've made exact matches for all of these. There's definitely some opportunity for creating some consistency here in a future PR.

**Comment by carkom** [comment]
@MatissJanis , review change requests are complete.

**Review by MatissJanis — APPROVED** [review]
Thank you for being so patient! And thanks for all the hard work on darkmode!
